src/test/java/org/opensearch/neuralsearch/query/NeuralQueryIT.java

```diff
@@ -377,6 +393,64 @@ public void testFilterQuery() {
         }
     }
 
+    @SneakyThrows
+    public void testQueryWithMaxDistance() {
+        String modelId = null;
+        try {
+            initializeIndexIfNotExist(TEST_BASIC_INDEX_NAME);
+            modelId = prepareModel();
+            NeuralQueryBuilder neuralQueryBuilder = new NeuralQueryBuilder(
+                TEST_KNN_VECTOR_FIELD_NAME_1,
+                TEST_QUERY_TEXT,
+                "",
+                modelId,
+                null,
+                100.0f,
+                null,
+                null,
+                null
+            );
+
+            Map<String, Object> searchResponseAsMap = search(TEST_BASIC_INDEX_NAME, neuralQueryBuilder, 1);
+            Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+            assertEquals("1", firstInnerHit.get("_id"));
+            float expectedScore = computeExpectedScore(modelId, testVector, TEST_SPACE_TYPE, TEST_QUERY_TEXT);
+            assertEquals(expectedScore, objectToFloat(firstInnerHit.get("_score")), DELTA_FOR_SCORE_ASSERTION);
+        } finally {
+            wipeOfTestResources(TEST_BASIC_INDEX_NAME, null, modelId, null);
+        }
+    }
+
+    @SneakyThrows
+    public void testQueryWithMinScore() {
```

Comment:
can we merge this one with a previous test method? In neural search we hav to be frugal with number of times we deploy model, it causes integ test failures in infra pipeline. If we reuse same model (making larger test methods with multiple setup and asserts) it works. For now there is no good long term solution for multiple deploy/undeploy model problem unfortunately.

Reply:
Thanks let me know the context, will update.